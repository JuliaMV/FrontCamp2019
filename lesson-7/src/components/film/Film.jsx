/* eslint-disable camelcase */
import React from 'react';
import PropTypes from 'prop-types';

import Logo from 'components/logo/Logo';

import css from './Film.module.css';

const Film = ({
  img, title, genres, date, vote_average, runtime, overview,
}) => (
  <div className={css.wrapper}>
    <div className={css.container}>
      <header className={css.header}>
        <Logo />
        searchBtn
      </header>
      <main className={css.content}>
        <img className={css.poster} src={img} alt="description" />
        <div>
          <div className={css.titleWrapper}>
            <h2 className={css.title}>{title}</h2>
            <span className={css.votes}>{vote_average}</span>
          </div>
          <span className={css.genres}>{genres.join(' ')}</span>
          <div className={css.meta}>
            <div className={css.year}>
              <span>{new Date(date).getFullYear()}</span>
              <span>yaer</span>
            </div>
            <div className={css.time}>
              <span>{runtime}</span>
              <span>min</span>
            </div>
          </div>
          <p>{overview}</p>
        </div>
      </main>
    </div>
  </div>
);

Film.propTypes = {
  title: PropTypes.string,
  genres: PropTypes.arrayOf(PropTypes.string),
  img: PropTypes.string,
  date: PropTypes.string,
  vote_average: PropTypes.number,
  runtime: PropTypes.number,
  overview: PropTypes.string,
};

Film.defaultProps = {
  title: 'Gemini',
  genres: [
    'Mystery',
    'Thriller',
  ],
  img: 'https://image.tmdb.org/t/p/w500/oIltQs7MPk7VQFG3DJfgC63mShU.jpg',
  date: '2018-03-30',
  vote_average: 10,
  runtime: 92,
  overview: 'A heinous crime tests the complex relationship between a tenacious personal assistant and her Hollywood starlet boss. As the assistant travels across Los Angeles to unravel the mystery, she must stay one step ahead of a determined policeman and confront her own understanding of friendship, truth and celebrity.',
};

export default Film;
