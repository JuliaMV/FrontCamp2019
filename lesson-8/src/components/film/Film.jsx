/* eslint-disable camelcase */
import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import Logo from 'components/logo/Logo';
import SearchIcon from 'components/searchIcon/searchIcon';

import css from './Film.module.css';

const Film = ({
  poster_path, title, genres, release_date, vote_average, runtime, overview,
}) => (
  <div className={css.wrapper}>
    <div className={css.container}>
      <header className={css.header}>
        <Logo />
        <Link to="/">
          <SearchIcon />
        </Link>
      </header>
      <main className={css.content}>
        <img className={css.poster} src={poster_path} alt="description" />
        <div>
          <div className={css.titleWrapper}>
            <h2 className={css.title}>{title}</h2>
            <span className={css.votes}>{vote_average}</span>
          </div>
          <p className={css.genres}>{genres.join(' ')}</p>
          <div className={css.meta}>
            <span className={css.year}>
              {new Date(release_date).getFullYear()}
            </span>
            <span className={css.time}>
              {`${runtime} min`}
            </span>
          </div>
          <p className={css.description}>{overview}</p>
        </div>
      </main>
    </div>
  </div>
);

Film.propTypes = {
  title: PropTypes.string.isRequired,
  genres: PropTypes.arrayOf(PropTypes.string).isRequired,
  poster_path: PropTypes.string.isRequired,
  release_date: PropTypes.string.isRequired,
  vote_average: PropTypes.number.isRequired,
  runtime: PropTypes.number.isRequired,
  overview: PropTypes.string.isRequired,
};

export default Film;
