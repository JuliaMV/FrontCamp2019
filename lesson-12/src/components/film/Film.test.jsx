import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import Film from './Film';

Enzyme.configure({ adapter: new Adapter() });


describe('Film component', () => {
  test('should render with props', () => {
    const props = {
      poster_path: 'http://image.com',
      title: 'News title',
      genres: ['fantasy'],
      release_date: '03.02.2019',
      vote_average: 15,
      runtime: 125,
      overview: 'Lorem ipsum',
      id: 1325,
    };
    // eslint-disable-next-line react/jsx-props-no-spreading
    const component = shallow(<Film {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
