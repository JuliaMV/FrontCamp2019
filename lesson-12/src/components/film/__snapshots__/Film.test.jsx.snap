// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film component should render with props 1`] = `
<div>
  <div>
    <header>
      <Logo />
      <Link
        to="/"
      >
        <SearchIcon />
      </Link>
    </header>
    <main>
      <img
        alt="description"
        src="http://image.com"
      />
      <div>
        <div>
          <h2>
            News title
          </h2>
          <span>
            15
          </span>
        </div>
        <p>
          fantasy
        </p>
        <div>
          <span>
            2019
          </span>
          <span>
            125 min
          </span>
        </div>
        <p>
          Lorem ipsum
        </p>
      </div>
    </main>
  </div>
</div>
`;
